build/ArtifactBuilder/Artifacts/LinuxPackage.cs

```diff
@@ -1,43 +1,187 @@
+                    var cpioFile = Path.Join(unpackedDir, new FileInfo(_packagePath).Name.Replace(".rpm", "") + ".cpio");
+                    UnpackWith7z(cpioFile, unpackedDir, true);
+                    return Path.Combine(unpackedDir, "usr", "local", "newrelic-dotnet-agent");
+                case "tar.gz":
+                    Directory.CreateDirectory(unpackedDir);
+                    var tar = UnGzip(_packagePath, unpackedDir);
+                    TarFile.ExtractToDirectory(tar, unpackedDir, true);
+                    return Path.Combine(unpackedDir, "newrelic-dotnet-agent");
+                default:
+                    throw new PackagingException($"Unknown extension {_fileExtension} for Linux package {_packageName}");
+            }
+        }
 
-            //Generate the checksum file based on the Linux standards for the checksum contents: <checksum_value> <checksum_mode><filename>
-            //<checksum_mode> is either a ' ' for text mode or '*' for binary mode.
-            File.WriteAllText($@"{OutputDirectory}\{fileInfo.Name}.sha256", $"{FileHelpers.GetSha256Checksum(packagePath)} *{fileInfo.Name}");
+        private string UnGzip(string gzipFile, string outputPath)
+        {
+            if (!gzipFile.EndsWith(GzipFileExtension))
+            {
+                throw new ArgumentException($"Input filename {gzipFile} does not end in '{GzipFileExtension}'");
+            }
+            var outputFile = Path.Join(outputPath, new FileInfo(gzipFile).Name.Replace(GzipFileExtension, ""));
+
+            using FileStream compressedFileStream = File.Open(gzipFile, FileMode.Open);
+            using FileStream outputFileStream = File.Create(outputFile);
+            using var decompressor = new GZipStream(compressedFileStream, CompressionMode.Decompress);
+            decompressor.CopyTo(outputFileStream);
+            return outputFile;
         }
+
+        private void UnpackWith7z(string archiveFile, string outputPath, bool extractWithFullPaths)
+        {
+            var pathTo7z = Get7zPath();
+            if (string.IsNullOrEmpty(pathTo7z))
+            {
+                throw new PackagingException("Unable to find path to 7z tool, can't validate contents of Linux archives.");
+            }
+
+            var operation = extractWithFullPaths ? "x" : "e";
+            var parameters = $"{operation} {archiveFile} -o{outputPath}";
+            var pInfo = new ProcessStartInfo(pathTo7z, parameters);
+            pInfo.CreateNoWindow = true;
+            pInfo.RedirectStandardOutput = true;
+            pInfo.RedirectStandardError = true;
+            pInfo.UseShellExecute = false;
+
+            var process = Process.Start(pInfo);
+            process.WaitForExit(30000);
+            if (!process.HasExited)
+            {
+                process.Kill();
+                throw new Exception($"7z unpacakge failed to complete in a timely fashion.");
```

Comment:
`Unpackage` is misspelled here and in another location.

Reply:
Good catch, thanks.